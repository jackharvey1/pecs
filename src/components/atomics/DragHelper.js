import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Helper = styled.div`
    position: absolute;
    width: 50%;
    height: 100%;
    z-index: 1;
    transform: translateX(+100%);
`;

const LeftHelper = styled.div`
    position: absolute;
    width: 50%;
    height: 100%;
    z-index: 1;
`;

const BigHelper = styled.div`
    position: relative;
    width: 100%;
    height: auto;
    align-self: stretch;
`;

const DragHelper = ({ shouldDisplayDragHelper, id, left, big }) => {
    if (shouldDisplayDragHelper) {
        if (left) {
            return <LeftHelper id={id} />;
        }

        if (big) {
            return <BigHelper id={id} />;
        }

        return <Helper id={id} />;
    }

    return null;
};

DragHelper.propTypes = {
    shouldDisplayDragHelper: PropTypes.bool,
    left: PropTypes.bool,
    big: PropTypes.bool,
    id: PropTypes.string.isRequired,
};

export default DragHelper;
