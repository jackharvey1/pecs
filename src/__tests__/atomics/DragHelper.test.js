import React from 'react';
import renderer from 'react-test-renderer';
import DragHelper from '../../components/atomics/DragHelper';

describe('DragHelper component', () => {
    it('renders a regular right translated helper when no prop is passed', () => {
        const wrapper = renderer.create(<DragHelper
            id="helper"
            shouldDisplay={true}
        />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a left translated helper when no prop is passed', () => {
        const wrapper = renderer.create(<DragHelper
            left
            id="helper"
            shouldDisplay={true}
        />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a large helper when prop big is passed', () => {
        const wrapper = renderer.create(<DragHelper
            big
            shouldDisplay={true}
            id="big-helper"
        />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('returns null when shouldDisplay is not true', () => {
        const wrapper = renderer.create(<DragHelper
            big
            shouldDisplay={false}
            id="big-helper"
        />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
