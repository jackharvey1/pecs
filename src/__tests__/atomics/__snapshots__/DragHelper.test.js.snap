// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragHelper component renders a large helper when prop big is passed 1`] = `
.c0 {
  position: relative;
  width: 100%;
  height: auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

<div
  className="c0"
  id="big-helper"
/>
`;

exports[`DragHelper component renders a left translated helper when no prop is passed 1`] = `
.c0 {
  position: absolute;
  width: 50%;
  height: 100%;
  z-index: 1;
}

<div
  className="c0"
  id="helper"
/>
`;

exports[`DragHelper component renders a regular right translated helper when no prop is passed 1`] = `
.c0 {
  position: absolute;
  width: 50%;
  height: 100%;
  z-index: 1;
  -webkit-transform: translateX(+100%);
  -ms-transform: translateX(+100%);
  transform: translateX(+100%);
}

<div
  className="c0"
  id="helper"
/>
`;

exports[`DragHelper component returns null when shouldDisplay is not true 1`] = `null`;
